### PR TITLE
Clang-Tidy adaptations

### PR DESCRIPTION
**Pull request by YunHsiao**
relevant: https://github.com/cocos-creator/engine/pull/8423

If auto-fix doesn't work and reports this error in the console, update to MSVC 2019 (16.9+):
```
...\intrin.h(56,1): error G7973E533: expected function body after function declarator [clang-diagnostic-error]
```
https://github.com/microsoft/STL/issues/1300

Also, the extension doesn't work in a multi-root workspace environment.

### CODE REVIEW

**Review by pandamicro — APPROVED** [review]

**Comment by pandamicro** [comment]
Clang tidy seems very strict, how do we apply the requirement ?

**Comment by minggo** [comment]
> Clang tidy seems very strict, how do we apply the requirement ? @minggo

It is caused by add `readability-*`. But if removing it, then readability-identifier-naming.XXX can not take effect. I do not know how to fix it, so we can just ignore the rules that are strict, such as add

```
Checks: >
- xxx,
-readability-uppercase-literal-suffix,
-readability-braces-around-statements
```

**Comment by minggo** [comment]
After some experiment, i think replace `readability-*` with `readability-identifier-*` is better.

**Comment by minggo** [comment]
I have changed to use readability-identifier-*. Please fix the coding style issue. If think some rule are not needed, then we can discuss it.

File: .clang-tidy [hunk 1]
```diff
@@ -1,81 +1,60 @@
...
+    - { key: readability-identifier-naming.ClassConstantCase,      value: UPPER_CASE }
+    - { key: readability-identifier-naming.ClassMemberCase,        value: camelBack  }
+    - { key: readability-identifier-naming.ProtectedMemberPrefix,  value: _          }
+    - { key: readability-identifier-naming.PrivateMemberPrefix,    value: _          }
+
+    - { key: readability-identifier-naming.EnumCase,               value: CamelCase  }
+    - { key: readability-identifier-naming.ScopedEnumConstantCase, value: UPPER_CASE }
+    - { key: readability-identifier-naming.EnumConstantCase,       value: UPPER_CASE }
 
+    - { key: readability-identifier-naming.FunctionCase,           value: aNy_CasE   }
```
Comment (minggo) [inline .clang-tidy:53]: Why function name is `aNy_CasE`?

File: cocos/renderer/gfx-gles2/GLES2PrimaryCommandBuffer.cpp [hunk 2]
```diff
@@ -38,15 +38,13 @@
 namespace cc {
 namespace gfx {
 
-GLES2PrimaryCommandBuffer::GLES2PrimaryCommandBuffer()
-: GLES2CommandBuffer() {
-}
+GLES2PrimaryCommandBuffer::GLES2PrimaryCommandBuffer() = default;
```
Comment (minggo) [inline cocos/renderer/gfx-gles2/GLES2PrimaryCommandBuffer.cpp:41]: Why not define it in .h?

Comment (YunHsiao) [reply cocos/renderer/gfx-gles2/GLES2PrimaryCommandBuffer.cpp:41]: https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-equals-default.html

File: .clang-tidy (hunk 1 above)
Comment (YunHsiao) [reply .clang-tidy:53]: Google uses the loose requirement on this. We have many functions that are not strictly following camelBack style, so I think for now we can hold on to this rule?

Comment (minggo) [reply .clang-tidy:53]: If using this rule, then `Class::Func()` will be valid?

Comment (YunHsiao) [reply .clang-tidy:53]: Yes

Comment (minggo) [reply .clang-tidy:53]: I am agree we can hold on this rule and add some comments for existing codes, but then we can not check new codes.

**Review by minggo — APPROVED** [review]

**Pull request merged**